engines/suppliers/db/migrate/20150825155330_create_order.rb

```diff
@@ -1,13 +1,35 @@
 class CreateOrder < ActiveRecord::Migration
-  def change
-    create_table :orders do |t|
-      t.integer :user_id, null: false
-      t.integer :group_id, null: false
+  def up
+    execute <<-SQL
+      CREATE TABLE orders (
+        id             serial  PRIMARY KEY,
+        from_user_id   integer REFERENCES users,
+        from_group_id  integer REFERENCES groups,
+        to_group_id    integer REFERENCES groups,
+        to_producer_id integer REFERENCES producers,
+        confirm_before timestamp without time zone NOT NULL,
+        created_at     timestamp without time zone NOT NULL,
+        updated_at     timestamp without time zone NOT NULL,
+        CHECK(
+          (
+            (from_user_id IS NOT NULL)::integer +
+            (from_group_id IS NOT NULL)::integer
+          ) = 1
+        ),
+        CHECK(
+          (
+            (to_group_id IS NOT NULL)::integer +
+            (to_producer_id IS NOT NULL)::integer
+          ) = 1
+        )
+      );
```

Comment:
@andresgutgon  Rails doesn't not support exclusive arcs, that's a better way to deal with polymorphism. Wherever I use it I put a link to more info in the migration. Read the doc on top of the file.

@sauloperez that's right, Rails doesn't keep track of custom queries. But that's not a concern IMHO. Rails doesn't offer a better way to deal with polymorphism. We cannot let Rails limit ourselves. There is no harm in using custom migrations, as usual you just need to know what you're doing.